server/proxy/pf_context.c

```diff
@@ -98,10 +103,21 @@ void pf_context_copy_settings(rdpSettings* dst, const rdpSettings* src, BOOL is_
 	dst->ClientRandomLength = before_copy->ClientRandomLength;
 	CopyMemory(dst->ClientRandom, before_copy->ClientRandom, before_copy->ClientRandomLength);
 
-	if (is_server)
+	if (is_dst_server)
 	{
 		free(dst->ServerCertificate);
 		dst->ServerCertificateLength = before_copy->ServerCertificateLength;
+
+		/* all of these should be marked as NULL, because they are already freed in the source
+		 * settings struct */
+		dst->TargetNetAddress = NULL;
```

Comment:
Just checked, if they are not `NULL` here some memory has been allocated in `freerdp_settings_copy` that needs to be cleaned up - or do I miss the point?

Reply:
no, they will be freed when context_free will be called in connection termination
they are set to null only in the destination struct, but the src struct contains those pointers and they will be freed